### PR TITLE
Fix convertToWorldSpace calculation issue

### PR DESCRIPTION
**Pull request by pandamicro**
Re: cocos-creator/fireball#4185

Changes proposed in this pull request:
- Fix convertToWorldSpace calculation issue

> Makes sure these boxes are checked before submitting your PR - thank you!
> - [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
> - [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
> - To official teams:
>   - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
>   - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/utils/base-node.js [hunk 1]
```diff
@@ -1574,8 +1574,10 @@ var BaseNode = cc.Class(/** @lends cc.Node# */{
             // ensure transform computed
             cc.director._visitScene();
         }
+        nodePoint.x -= this._anchorPoint.x * this._contentSize.width;
+        nodePoint.y -= this._anchorPoint.y * this._contentSize.height;
```
Comment (jareguo) [inline cocos2d/core/utils/base-node.js]: 不应该修改传入坐标吧？

Comment (pandamicro) [reply cocos2d/core/utils/base-node.js]: 嗯，有道理

File: cocos2d/core/utils/base-node.js [hunk 2]
```diff
@@ -1574,8 +1574,10 @@ var BaseNode = cc.Class(/** @lends cc.Node# */{
             // ensure transform computed
             cc.director._visitScene();
         }
-        var worldPositionIgnoreAnchorPoint = this._sgNode.convertToWorldSpace(nodePoint);
-        return cc.pSub(worldPositionIgnoreAnchorPoint, cc.p(this._anchorPoint.x * this._contentSize.width, this._anchorPoint.y * this._contentSize.height));
+        var x = nodePoint.x - this._anchorPoint.x * this._contentSize.width;
+        var y = nodePoint.y - this._anchorPoint.y * this._contentSize.height;
+        var worldPositionIgnoreAnchorPoint = this._sgNode.convertToWorldSpace(cc.v2(x, y));
+        return worldPositionIgnoreAnchorPoint;
```
Comment (jareguo) [inline cocos2d/core/utils/base-node.js:1580]: 我觉得 worldPositionIgnoreAnchorPoint 这个变量可以不需要了吧

Comment (jareguo) [inline cocos2d/core/utils/base-node.js:1580]: 因为已经计算了 anchor 了，所以现在不是 ignoreAnchorPoint 了

Comment (pandamicro) [reply cocos2d/core/utils/base-node.js:1580]: 是 ignore 啊，transform 中的坐标是包含了 anchor point 的，在这里去掉了 Anchor point 获得原点坐标，然后经过坐标转换，就变成了 worldPosition ignore anchor point 了。

Comment (jareguo) [reply cocos2d/core/utils/base-node.js:1580]: 嗯，之前变量名没写对，现在反过来刚好可以了。

**Comment by jareguo** [comment]
LGTM

**Comment by zilongshanren** [comment]
Also fixed this issue:  https://github.com/cocos-creator/fireball/issues/4124

**Pull request merged**